packages/runtime/container-runtime/src/gc/gcHelpers.ts

```diff
@@ -26,10 +27,63 @@ import {
 	GCFeatureMatrix,
 	GCVersion,
 	IGCMetadata,
+	IGCResult,
 	runSweepKey,
 	throwOnTombstoneLoadKey,
 	throwOnTombstoneUsageKey,
 } from "./gcDefinitions";
+import {
+	IGarbageCollectionNodeData,
+	IGarbageCollectionSnapshotData,
+	IGarbageCollectionState,
+	IGarbageCollectionSummaryDetailsLegacy,
+} from "./gcSummaryDefinitions";
+
+/**
+ * Runs garbage collection on the given reference graph.
+ * @param referenceGraph - The reference graph to run GC on. It's a list of nodes where each node has an id and set of
+ * routes to other nodes in the graph.
+ * @param rootIds - The ids of root nodes that are considered referenced.
+ * @returns the ids of referenced nodes and the ids of deleted nodes in the referenced graph.
+ */
+export function runGarbageCollection(
```

Comment:
As mentioned in person and above, I'm curious if we can group all the graph algorithm code/helpers together to make it really clear the scope of that code.

Reply:
Moved this to a separate file. We can move the other graph algorithm stuff can move here later on opportunistically.